digest.go

```diff
@@ -0,0 +1,295 @@
+package resty
```

Comment:
@segevda Did you use any code reference or documentation for this implementation? If yes, please add those references in the code file.

Reply:
This code was loosely inspired by [bobziuchkovski/digest](https://github.com/bobziuchkovski/digest). I don't mind referencing it, just not sure if this has any license implications. The code in that repo is under Apache 2.0 and resty is under MIT. WDYT?

Reply:
I see, thanks for the info.